### DISCUSSION TITLE
Vue packages version mismatch

### DISCUSSION THREAD

**Discussion by newcat** [newcat/baklavajs#247] [Q&A] [answered]
Hey Guys,

I cloned the demo and tried to run it with Vue3 following the instructions above. `rpm run dev` produces the following error:

` ERROR  Failed to compile with 1 errors 
 error  in ./src/App.vue
Module build failed: Error: 
Vue packages version mismatch:
- vue@2.7.10 (/Users/djeetee/Documents/github repos/BaklavaJSv1.xDemo/node_modules/vue/dist/vue.runtime.common.js)
- vue-template-compiler@2.7.14 (/Users/djeetee/Documents/github repos/BaklavaJSv1.xDemo/node_modules/vue-template-compiler/package.json)`

Any thoughts on how to fix this? it could be my lack of understanding of vue3 as I just started working with it.

Thank you in advance.

g.

_Originally posted by @djeetee in https://github.com/newcat/baklavajs/issues/135#issuecomment-1397844599_

**Comment by newcat** [comment]
Hey @djeetee, I moved your comment here since it didn't really fit into the other issues. No worries though.

To me, it looks like you are running a Vue 2 project (https://github.com/djeetee/BaklavaJSv1.xDemo). But you are also using the Vue 2 version of Baklava so it looks fine to me and I can also run your CodeSandbox. Can you confirm the issue still exists?

**Reply by djeetee** [reply]
Hey @newcat, thank you for the reply.

I cloned the demo (the one here https://codesandbox.io/s/baklavajs-example-jyc6f?file=/src/App.vue&ref=madewithvuejs.com) locally, ran rpm install then rpm run dev hoping i can run it with Vue3

then i realized (wrongly ??) that I needed to update main.js following your instructions. I did but then I got the errors i mentioned in my previous post. btw, I also ran npm install @baklavajs/core @baklavajs/plugin-options-vue3 @baklavajs/plugin-renderer-vue3 

From your reply I'm assuming the last command installed Baklava v2 hence the issue.

To make things clearer, what I would like to do is to get a demo (preferably of baklava v2) working with Vue3 so that I could experiment with it to see if I can be used in an app I'm putting together.

Is there a Baklava V2-specific demo or just the V1 one at the moment?

I hope this makes things clearer.

Thanks again.

**Reply by newcat** [reply]
I have now created a v2 demo: https://codesandbox.io/s/baklavajs-v2-example-zpfkec?file=/src/App.vue

**Reply by djeetee** [reply]
Amazing! I will give it a try tomorrow. Thank you so much.

**Reply by djeetee** [reply]
Not my lucky week! `yarn init` works ok. `yarn serve` give me this:

`node:internal/crypto/hash:71
  this[kHandle] = new _Hash(algorithm, xofLen);
Error: error:0308010C:digital envelope routines::unsupported
blah blah
`
which seems to be related to the version of node. I have the latest (19.4.0) and due is version 3.2.45

Any thoughts?

what version of node do you have?

**Reply by djeetee** [reply, marked as answer]
a quick update.

I created a brand new Vu3 project. I use Vita btw.
Cleaned it up from the standard install files & components.
Copied the source files from your v2 demo.

then...

`npm install`

`vite`

it seems that did the job. I can launch the project and i can see the nodes editor.

Yay!

I'm hoping I could use Baklava in my project. Time to experiment.

Thank you again.
